plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -167,16 +174,25 @@ public CloseableIterator<List<Object>> iterator()
             addCloseable(planFilesIterator);
 
             return new CloseableIterator<>() {
+                private Iterator<DeleteFile> deleteFileIterator = emptyIterator();
+
                 @Override
                 public boolean hasNext()
                 {
-                    return !closed && planFilesIterator.hasNext();
+                    return !closed && (planFilesIterator.hasNext() || deleteFileIterator.hasNext());
                 }
 
                 @Override
                 public List<Object> next()
                 {
-                    return getRecord(planFilesIterator.next().file());
+                    if (deleteFileIterator.hasNext()) {
+                        return getRecord(deleteFileIterator.next());
+                    }
+                    else {
+                        FileScanTask planFileTask = planFilesIterator.next();
+                        deleteFileIterator = planFileTask.deletes().iterator();
```

Comment:
This has the potential to include duplicates. Equality deletes can belong to several files, so those will show up multiple times here.

Reply:
Good point. Let me find a way to deduplicate the delete files and add a test case for this.

Reply:
I think by using the metadata table scan suggested by @findinpath, it can avoid duplicated equality files in the final results. I have added one test in `TestIcebergV2`.

Reply:
@0xffmeta  try the following in a product test environment

```
testing/bin/ptl env up  --environment 'singlenode-spark-iceberg'  --config 'config-default'
```


via `spark-sql` 

```
create table default.test1 (x int) using iceberg tblproperties ('format-version' = '2','write.delete.mode' = 'merge-on-read', 'write.update.mode'='merge-on-read',  'write.merge.mode'='merge-on-read');
```

via `trino`
```
insert into t1 values 1,2,3,4,5,6;
insert into t1 values 1,2,3,4,5,6;
```

via `spark-sql`
```
delete from t1 where x=2; -- position deletes
delete from t1 where x > 2; -- position deletes
```

check via Trino whether `$files` returns duplicated equality delete files.

via `flink` can be written equality deletes
```
delete from t1 where x=2;
```
The equality deletes may be duplicated (in case they apply to several data files) in the content of `$files` - in such a case we'll need to figure out a way to deduplicate such entries...

Reply:
Great. Thanks for the detailed steps. Let me try to check this in the test env.

Reply:
> check via Trino whether $files returns duplicated equality delete files.
The trino result:
<img width="1776" alt="image" src="https://user-images.githubusercontent.com/98149057/225241895-731fb770-e8f0-4958-b38f-4e71c5cedfcf.png">

Reply:
> via flink can be written equality deletes

Not sure which environment should be used to create a flink-sql container... @findinpath Do you know how to run flink sql in product test env?